src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+            putFieldError(errorPath, CUKFSKeyConstants.ERROR_DOCUMENT_ACCT_GLB_MAINT_ACCOUNT_CLOSED_NO_FUND_BALANCES, new String[]{detail.getAccountNumber()});
+            success &= false;
+        }
+
+        // We must not have any pending labor ledger entries
+        if (SpringContext.getBean(LaborModuleService.class).hasPendingLaborLedgerEntry(detail.getAccount().getChartOfAccountsCode(), detail.getAccount().getAccountNumber())) {
+            putFieldError(errorPath, CUKFSKeyConstants.ERROR_DOCUMENT_ACCT_GLB_MAINT_ACCOUNT_CLOSED_PENDING_LABOR_LEDGER_ENTRIES, new String[]{detail.getAccountNumber()});
+            success &= false;
+        }
+
+        return success;
+    }
+
+    /**
+     * This method checks to see if the account expiration date is today's date or earlier
+     *
+     * @param newAccount
+     * @return fails if the expiration date is null or after today's date
+     */
+    protected boolean checkAccountExpirationDateValidTodayOrEarlier(CuAccountGlobal newAccount) {
+
+        // get today's date, with no time component
+        Date todaysDate = new Date(getDateTimeService().getCurrentDate().getTime());
+        todaysDate.setTime(DateUtils.truncate(todaysDate, Calendar.DAY_OF_MONTH).getTime());
+        // TODO: convert this to using Wes' Kuali KfsDateUtils once we're using Date's instead of Timestamp
+
+        // get the expiration date, if any
+        Date expirationDate = newAccount.getAccountExpirationDate();
+        if (ObjectUtils.isNull(expirationDate)) {
+            putFieldError(KFSPropertyConstants.ACCOUNT_EXPIRATION_DATE, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ACCT_CANNOT_BE_CLOSED_EXP_DATE_INVALID);
+            return false;
+        }
+
+        // when closing an account, the account expiration date must be the current date or earlier
+        expirationDate.setTime(DateUtils.truncate(expirationDate, Calendar.DAY_OF_MONTH).getTime());
+        if (expirationDate.after(todaysDate)) {
+            putFieldError(KFSPropertyConstants.ACCOUNT_EXPIRATION_DATE, KFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ACCT_CANNOT_BE_CLOSED_EXP_DATE_INVALID);
+            return false;
+        }
+
+        return true;
+    }
+    
+    protected boolean checkSubFundGroup() {    	
+		boolean success = true;
+		for (AccountGlobalDetail detail : newAccountGlobal.getAccountGlobalDetails()) {
+			success &= checkSubFundGroup(detail);
+		}
+		return success;    	
+    }
+    
+    
+    /**
+     * This method checks to see if any SubFund Group rules were violated Specifically: if SubFundGroup is empty or not "PFCMR" we
+     * cannot have a campus code or building code if SubFundGroup is "PFCMR" then campus code and building code "must" be entered
+     * and be valid codes
+     *
+     * @param maintenanceDocument
+     * @return false on rules violation
+     */
+    protected boolean checkSubFundGroup( AccountGlobalDetail detail) {
```

Comment:
Lots of "!StringUtils.isBlank()" checks are in this method and should ideally be replaced with "StringUtils.isNotBlank()"; only a few are noted below.